### PR TITLE
ESP Asic Memory Integration Flow Stratus Compliant

### PR DESCRIPTION
**Pull request by maico-cassel**

### CODE REVIEW

File: .gitignore
Comment (jzuckerman) [inline .gitignore:1]: why did you add these to the gitignore? To avoid accidentally pushing technology files? why the flists also?

File: accelerators/stratus_hls/common/hls/project.tcl [hunk 1]
```diff
@@ -45,11 +48,14 @@ if {[lsearch $fpga_techs $TECH] >= 0} {
     set TECH_IS_XILINX 1
 
 }
-if {[lsearch $asic_techs $TECH] >= 0} {
-    set_attr verilog_files "$ESP_ROOT/rtl/sim/$TECH/verilog/*v $ESP_ROOT/rtl/techmap/$TECH/mem/*v"
+if {[lsearch $asic_techs $TECH_TYPE] >= 0} {
```
Comment (jzuckerman) [inline accelerators/stratus_hls/common/hls/project.tcl:51]: TECH_TYPE is always "asic" now? and then TECH is the particular technology? Because I see we are using TECH to set the frequency/input delays

File: accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#  Copyright (c) 2011-2023 Columbia University, System Level Design Group
-#  SPDX-License-Identifier: Apache-2.0
```
Comment (jzuckerman) [inline accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl:1]: can you add back the copyright?

File: rtl/techmap/asic/clkgen_gf12.vhd [hunk 3]
```diff
@@ -1,4 +1,4 @@
--- Copyright (c) 2011-2023 Columbia University, System Level Design Group
+-- Copyright (c) 2011-2022 Columbia University, System Level Design Group
 -- SPDX-License-Identifier: Apache-2.0
```
Comment (jzuckerman) [inline rtl/techmap/asic/clkgen_gf12.vhd:2]: fix copyright

File: rtl/techmap/asic/pads_gf12.vhd [hunk 4]
```diff
@@ -1,4 +1,4 @@
--- Copyright (c) 2011-2023 Columbia University, System Level Design Group
+-- Copyright (c) 2011-2022 Columbia University, System Level Design Group
 -- SPDX-License-Identifier: Apache-2.0
```
Comment (jzuckerman) [inline rtl/techmap/asic/pads_gf12.vhd:2]: fix copyright

File: utils/flist/techmap_vhdl.flist [hunk 5]
```diff
@@ -1,39 +0,0 @@
...
-techmap/virtexup/memory_unisim.vhd
-techmap/virtexup/sysmon_unisim.vhd
-techmap/virtexup/buffer_unisim.vhd
-techmap/virtexup/spictrl_unisim.vhd
-techmap/virtexup/pads_unisim.vhd
-techmap/virtexup/tap_unisim.vhd
-techmap/virtexup/clkgen_unisim.vhd
-techmap/virtexup/sgmii_vcu118.vhd
-techmap/virtexup/sgmii_vcu118_clock_reset.vhd
-techmap/virtexup/sgmii_vc707.vhd
```
Comment (jzuckerman) [inline utils/flist/techmap_vhdl.flist:39]: why are you deleting FPGA files?

File: utils/flist/techmap_vlog.flist [hunk 6]
```diff
@@ -1,123 +0,0 @@
...
-techmap/virtexup/sram_b/unisim_sram_b_13abits.v
-techmap/virtexup/sram_b/unisim_sram_b_9abits.v
-techmap/virtexup/sram_b/unisim_sram_b_8abits.v
-techmap/virtexup/sgmii_vcu118_reset_sync.v
-techmap/virtexup/mem/BRAM_1024x16.v
-techmap/virtexup/mem/BRAM_16384x1.v
-techmap/virtexup/mem/BRAM_2048x8.v
-techmap/virtexup/mem/BRAM_4096x4.v
-techmap/virtexup/mem/BRAM_512x32.v
-techmap/virtexup/mem/BRAM_8192x2.v
```
Comment (jzuckerman) [inline utils/flist/techmap_vlog.flist:123]: why deleting FPGA files?

**Review by jzuckerman — COMMENTED** [review]
Looks good. Just a few questions and minor things to fix

File: accelerators/stratus_hls/common/hls/project.tcl (hunk 1 above)
Comment (maico-cassel) [reply accelerators/stratus_hls/common/hls/project.tcl:51]: Yes. For asic flow, TECH_TYPE is always asic. This make ESP repo with only "asic" files instead of having files foe each asic technology.
TECH is for the particular technology to keep similar structure inside esp/tech folder. Therefore, for a given asic technology, every HLS generated accelerator will have its specific technology folder and would avoid future mistakes in a project with tight deadline.

File: accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl (hunk 2 above)
Comment (maico-cassel) [reply accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl:1]: I haven't removed it. I will add them back anyways. There was a command to do/update them automatically, right!?

File: rtl/techmap/asic/clkgen_gf12.vhd (hunk 3 above)
Comment (maico-cassel) [reply rtl/techmap/asic/clkgen_gf12.vhd:2]: I haven't changed this file

File: rtl/techmap/asic/pads_gf12.vhd (hunk 4 above)
Comment (maico-cassel) [reply rtl/techmap/asic/pads_gf12.vhd:2]: I haven't changed this file

File: utils/flist/techmap_vlog.flist (hunk 6 above)
Comment (maico-cassel) [reply utils/flist/techmap_vlog.flist:123]: I am not deleting them. techmap_v*.flist is now automatically generated once "check_all_srcs" target inside rtl.mk is called. This is required due to different memory wrapper names according to the user size preference.

File: utils/flist/techmap_vhdl.flist (hunk 5 above)
Comment (maico-cassel) [reply utils/flist/techmap_vhdl.flist:39]: I am not deleting them. techmap_v*.flist is automatically generated. techmap_vhdl.flist calls pads and clock generator, something I still haven't worked on. Perhaps this file won't be required to be generated as techmap_vlog.flist does.

File: accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl (hunk 2 above)
Comment (jzuckerman) [reply accelerators/stratus_hls/sort_stratus/hw/hls/project.tcl:1]: KL has a script that does all the files, but I think since you have just 3, you can do it manually

File: .gitignore
Comment (jzuckerman) [reply .gitignore:1]: I think I'd prefer to have these in .gitignores further down in the hierarchy? 
Can we create new .gitignores in:
utils/flist/
rtl/techmap/asic/
rtl/sim/gf12/

Comment (maico-cassel) [reply .gitignore:1]: I don't like the idea of having many .gitignores in the folder hierarchy. It would be a nightmare to find what we are ignoring in the future.
